driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/OperationExecutorImpl.java

```diff
@@ -62,31 +65,38 @@
             session.notifyOperationInitiated(operation);
         }
 
-          return clientSessionHelper.withClientSession(session, this)
-                .map(clientSession -> getReadWriteBinding(readPreference, readConcern, clientSession,
-                                                              session == null && clientSession != null))
-                .switchIfEmpty(Mono.fromCallable(() -> getReadWriteBinding(readPreference, readConcern, session, false)))
-                .flatMap(binding -> {
-                    if (session != null && session.hasActiveTransaction() && !binding.getReadPreference().equals(primary())) {
-                        binding.release();
-                        return Mono.error(new MongoClientException("Read preference in a transaction must be primary"));
-                    } else {
-                        return Mono.<T>create(sink -> operation.executeAsync(binding, (result, t) -> {
-                            try {
+        return new Publisher<T>() {
+            @Override
+            public void subscribe(final Subscriber<? super T> s) {
+                // TODO: Extract a request context from the subscriber and add it to the binding
```

Comment:
This is where we would plug in the request context provider to extract it from the Subscriber.  I wasn't sure how to do this without introducing a custom Publisher that intercepts the subscribe method.

Reply:
hey @jyemin. before going into complex solution like writing ad-hoc publishers, have you explored built-in operators like `deferContextual` (for a source-like publisher) or `transformDeferredContextual` (for an intermediate step)? Both of these aim at exposing the `ContextView` of the downstream `Subscriber`.

Another more advanced solution to explore is the usage of `Operators.lift`, but it is generally used with hooks rather than directly in chains, where it doesn't have much over `transformDeferredContextual` (since it would be used through `transform`), other than giving access to the whole `Subscriber`.

Reply:
@simonbasle I don't think this changes the public API, and given that we're still on Californium I propose to leave this as is for now.